### PR TITLE
mm_np binding

### PR DESCRIPTION
**Pull request by segalmax**
Contains various updates and implementations that are all related to binding the mm_np C structure.

### CODE REVIEW

File: py_bind/optv/parameters.pyx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        
+    def get_n1(self):
+        return self._mm_np[0].n1
+    
+    def set_n1(self, n1):
+        self._mm_np[0].n1 = n1
+        
+    def get_n2(self):
+        list_size = sizeof(self._mm_np[0].n2) / sizeof(self._mm_np[0].n2[0])
+        n2_list = [None] * list_size
```
Comment (yosefm) [inline py_bind/optv/parameters.pyx]: Should use a numpy array, not a list. Many actions which require a loop for a list have an inbuild invisible loop in a numpy array.

File: py_bind/test/test_MultimediaParams.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import unittest
+from optv.parameters import MultimediaParams
+
+class Test_MultimediaParams(unittest.TestCase):
+    
+    @staticmethod
+    def identical_lists(self, list_1, list_2):
```
Comment (yosefm) [inline py_bind/test/test_MultimediaParams.py]: You can compare lists directly in Python, and the comparison operator basically implements your logic. Note this IPython session:
```
In [1]: [1, 2, 3] == [1, 2, 3]
Out[1]: True

In [2]: [1, 2, 3] == [1, 2, 3.]
Out[2]: True

```

File: py_bind/test/test_MultimediaParams.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        d_list = [55,66,77]
+        
+        m = MultimediaParams(nlay=3, n1=2, n2=n2_list, d=d_list, n3=4, lut=1)
+        
+        self.failUnlessEqual(m.get_nlay(), 3)
+        self.failUnlessEqual(m.get_n1(), 2)
+        self.failUnlessEqual(m.get_n3(), 4)
+        self.failUnlessEqual(m.get_lut(), 1)
+        
+        self.assertTrue(Test_MultimediaParams.identical_lists(self, m.get_d(), d_list))
```
Comment (yosefm) [inline py_bind/test/test_MultimediaParams.py]: I suggest numpy.testing.assert_array_equal()

File: py_bind/test/test_MultimediaParams.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        self.assertTrue(Test_MultimediaParams.identical_lists(self, m.get_n2(), n2_list))
+        
+        self.failUnlessEqual(m.__str__(), "nlay=\t3 \nn1=\t2.0 \nn2=\t{11.0, 22.0, 33.0} \nd=\t{55.0, 66.0, 77.0} \nn3=\t4.0 \nlut=\t1 ")
```
Comment (yosefm) [inline py_bind/test/test_MultimediaParams.py]: Is this line still needed?

**Comment by yosefm** [comment]
Final comment on PR: it does not compile because of two problems:
1. redefinition of `compare_control_par()`
2. Too few arguments to `strncmp()`
Make it compile before you submit!

**Comment by alexlib** [comment]
is it ready for the testing?

**Comment by alexlib** [comment]
Alexs-MacBook-Pro:py_bind alex$ python test/test_MultimediaParams.py -v
test_mm_np_instantiation (__main__.Test_MultimediaParams) ... ok

----------------------------------------------------------------------
Ran 1 test in 0.001s

OK

**Comment by yosefm** [comment]
haven't tested the latest commits, but should be ok.

**Comment by alexlib** [comment]
I tested. see above. of course it's a single and simple test, but seems to be fine. Let's accept and move forward.

**Comment by yosefm** [comment]
Added directly to upstream. Good job.

**Pull request closed without merging**